ports/nrf/modules/machine/temp.c

```diff
@@ -88,7 +88,11 @@ STATIC mp_obj_t machine_temp_read(mp_uint_t n_args, const mp_obj_t *args) {
     }
 #endif // BLUETOOTH_SD
 
-    return MP_OBJ_NEW_SMALL_INT(nrf_temp_read());
+    nrf_temp_task_trigger(NRF_TEMP, NRF_TEMP_TASK_START);
```

Comment:
Before `nrf_temp_task_trigger()`, we should probably clear the `DATARDY`, in case it is already set when we enter the function.
`nrf_temp_event_clear(NRF_TEMP, NRF_TEMP_EVENT_DATARDY);`